pkg/instancegroups/instancegroups.go

```diff
@@ -221,6 +236,80 @@ func (r *RollingUpdateInstanceGroup) RollingUpdate(rollingUpdateData *RollingUpd
 	return nil
 }
 
+func (r *RollingUpdateInstanceGroup) maybeTaintAllNeedUpdate(update []*cloudinstances.CloudInstanceGroupMember, rollingUpdateData *RollingUpdateCluster, noneReady bool, uIdx int) error {
+	if r.CloudGroup.InstanceGroup.Spec.Role != api.InstanceGroupRoleNode || rollingUpdateData.CloudOnly {
+		return nil
+	}
+
+	if noneReady {
+		// Wait until after one node is deleted and its replacement validates before the mass-cordoning
+		// in case the current spec does not result in usable nodes.
+		if uIdx != 1 || len(update) < 2 {
+			return nil
+		}
+	} else {
+		if uIdx != 0 || len(update) < 1 {
+			return nil
+		}
+	}
+
+	var toTaint []*corev1.Node
+	for _, u := range update {
+		if u.Node != nil && !u.Node.Spec.Unschedulable {
+			foundTaint := false
+			for _, taint := range u.Node.Spec.Taints {
+				if taint.Key == rollingUpdateTaintKey {
+					foundTaint = true
+				}
+			}
+			if !foundTaint {
+				toTaint = append(toTaint, u.Node)
+			}
+		}
+	}
+	if len(toTaint) > 0 {
+		noun := "nodes"
+		if len(toTaint) == 1 {
+			noun = "node"
```

Comment:
It's normally fine not to worry about this, or just do `node(s)`, but this does look better!